apps/admin-gui/src/app/shared/components/dialogs/applications-list-columns-change-dialog/applications-list-columns-change-dialog.component.ts

```diff
@@ -74,32 +100,74 @@ export class ApplicationsListColumnsChangeDialogComponent implements OnInit {
               this.attribute = attribute;
               const configuredColumns: string[] = attribute.value as string[];
               if (configuredColumns !== null && configuredColumns.length > 0) {
-                this.selectedColumns.setValue(configuredColumns);
+                this.selectedColumns = this.friendlyToDisplayAttrNames(configuredColumns);
               } else {
-                this.selectedColumns.setValue(this.simpleColumns);
+                this.selectedColumns = this.simpleColumns;
               }
             });
+          this.loading = false;
         });
     }
   }
 
   confirm(): void {
+    if (
+      this.selectedColumns === null ||
+      this.selectedColumns.length === 0 ||
+      containsExactlyInAnyOrder(this.selectedColumns, this.simpleColumns)
+    ) {
+      this.attribute.value = [];
+    }
     if (this.data.groupId) {
       this.changeGroupAttribute();
     } else {
       this.changeVoAttribute();
     }
   }
+
   cancel(): void {
     this.dialogRef.close(false);
   }
   default(): void {
-    this.attribute.value = [];
-    this.confirm();
+    this.selectedColumns = [...this.simpleColumns];
+  }
+
+  template(): void {
+    this.selectedColumns = this.friendlyToDisplayAttrNames(
+      this.templateAttribute.value as string[]
+    );
+  }
+
+  toggleEvent(names: string[]): void {
+    // Replace array in-place so it won't trigger ngOnChanges
+    this.selectedColumns.splice(0, this.selectedColumns.length, ...names);
+  }
+
+  refreshOnClosed(): void {
+    this.selectedColumns = [...this.selectedColumns];
+  }
+
+  private displayToFriendlyAttrNames(columns: string[]): string[] {
```

Comment:
You can join columnTranslation with something like these 2 functions and pass it into [mainStringFunction] of the string search select. That should handle displaying of 'displayName' without having to do awkward conversions

Reply:
good idea, the conversions were really terrible.